grakn/rpc/cluster/replica_info.py

```diff
@@ -41,22 +41,30 @@ def primary_replica(self) -> Optional["ReplicaInfo.Replica"]:
         primaries = [replica for replica in self._replicas.values() if replica.is_primary()]
         return max(primaries, key=lambda r: r.term) if primaries else None
 
+    def preferred_secondary_replica(self) -> "ReplicaInfo.Replica":
+        return next(iter([replica for replica in self._replicas.values() if replica.is_preferred_secondary()]), next(iter(self._replicas.values())))
```

Comment:
@alexjpwalker type annotation indicates you're returning one thing but it looks like you're returning a tuple. Could you elaborate on what's supposed to be returned?

Reply:
It's not a tuple, it's: "take the first replica that is a preferred secondary, if it exists; else, just take the first replica"

Reply:
@alexjpwalker 

>It's not a tuple

So you're saying it returns just one instance?

>take the first replica that is a preferred secondary, if it exists

But if it doesn't exist we'll do `next(iter([]))` which will raise `StopIteration`

I think we could rewrite it for clarity too:

```
for replica in self._replicas.values():
    if replica.is_preferred_secondary():
        return replica
return next(iter(self._replicas.values())))
```

Reply:
> But if it doesn't exist we'll do next(iter([])) which will raise StopIteration

This isn't true - if you do:

`next([], 1)`

it returns `1`, it doesn't throw.

Reply:
And yes, it's just a `next` call so it always returns a single instance.

Reply:
@alexjpwalker got it! it's a **single** `next` call, second argument to which is **also** a `next` call. What a Python riddle!